website/addons/dmptool/model.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+
+from modularodm import fields
+
+from framework.exceptions import HTTPError
+
+from website.addons.base import (
+    AddonOAuthNodeSettingsBase, AddonOAuthUserSettingsBase, exceptions,
+)
+from website.addons.base import StorageAddonBase
+
+from website.addons.dmptool.serializer import DmptoolSerializer
+
+ROOT_FOLDER_NAME = '/'
+
+
+class DmptoolProvider(object):
+    """An alternative to `ExternalProvider` not tied to OAuth"""
+
+    name = 'Dmptool'
+    short_name = 'dmptool'
+    serializer = DmptoolSerializer
+
+    def __init__(self, account=None):
+        super(DmptoolProvider, self).__init__()
+        # provide an unauthenticated session by default
+        self.account = account
+
+    def __repr__(self):
+        return '<{name}: {status}>'.format(
+            name=self.__class__.__name__,
+            status=self.account.provider_id if self.account else 'anonymous'
+        )
+
+class AddonDmptoolUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+class AddonDmptoolNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+    oauth_provider = DmptoolProvider
+    serializer = DmptoolSerializer
+
+    folder_id = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_name = fields.StringField(default=ROOT_FOLDER_NAME)
+    folder_path = fields.StringField(default=ROOT_FOLDER_NAME)
```

Comment:
About [creating a custom fangorn configuration](https://hyp.is/WjGxiBotEeeCqm-ggdfuew/github.com/CenterForOpenScience/osf.io/pull/7030): you [point to the OneDrive PR](https://github.com/CenterForOpenScience/osf.io/pull/7030#discussion_r109745451) as an example.

To get started with configuring Fangorn for the DMPTool addon, do I copy `website/addons/onedrive/static/onedriveFangornConfig.js` to `website/addons/dmptool/static/dmptoolFangornConfig.js` and then do a bit of search & replace s/onedrive/dmptool/g.  I'll also have to create a new `website/addons/dmptool/static/files.js` containing

```
require('./dmptoolFangornConfig.js');
```

Do I have to do anything else for `dmptool/static/files.js` to be picked up?